daemon/service-db.cc

```diff
@@ -370,12 +488,16 @@ MLServiceDB::delete_model (const std::string name)
   std::string key_with_prefix = DB_KEY_PREFIX;
   key_with_prefix += name;
 
-  sql = g_strdup_printf ("DELETE FROM tblModel WHERE key = '%s';",
+  if (0U == version)
+    sql = g_strdup_printf ("DELETE FROM tblModel WHERE key = '%s';",
       key_with_prefix.c_str ());
+  else
+    sql = g_strdup_printf ("DELETE FROM tblModel WHERE key = '%s' and version = %u;",
+      key_with_prefix.c_str (), version);
```

Comment:
what about checking the existence of the specific version before executing the delete query?

Reply:
Added existence check, PTAL.